repos/system_upgrade/el7toel8/actors/targetuserspacecreator/libraries/userspacegen.py

```diff
@@ -130,15 +133,22 @@ def _create_target_userspace_directories(target_userspace):
         )
 
 
-def gather_target_repositories(target_rhsm_info):
+def gather_target_repositories():
     """
     Performs basic checks on requirements for RHSM repositories and returns the list of target repository ids to use
     during the upgrade.
     """
+    target_rhsm_repoids = []
+    base = yum.YumBase()
```

Comment:
@vinzenz, @pirat89, can I execute this in the nspawn container? Do I need to use the context somehow?

Reply:
Not like this, if you want to execute it in the nspawn container you need to make it an external script and use the context.run function to run it.
Basically you'd need to move the yum part in a python script and print the output so it can be parsed by the actor, put the script in the tools folder of the actor and run the script with context.run and parse the stdout

Reply:
Thanks, @vinzenz. I feared it could be this kind of cumbersome. I wanted to do it more elegantly using yum python module, but in this case I think I'll just call `yum repoinfo` which also includes the information I need.